### PR TITLE
Fix installation without websrv

### PR DESCRIPTION
**Pull request by bemyak**
This PR fixes installation with `nextcloud_install_websrv: false` option. Currently, `php_install` always installs the webserver without taking the option into account.

### CODE REVIEW

**Comment by staticdev** [comment]
@bemyak could you give a bit more context explaining what you changed? Did you test it also tour changed with ’nextcloud_install_websrv: true'?

**Comment by bemyak** [comment]
> Could you give a bit more context explaining what you changed?

Sure. In general, I'm trying to make this role work with Caddy web server, which is installed as a separate role.

I've extracted the web server installation from `php-install.yml` into a separate file `websrv_install.yml`. This file is only applied when `nextcloud_install_websrv` is set to true.

Another minor change is the explicit installation of `php-fpm`. I believe currently it's installed as a web server dependency, but when there is none, it should be installed as part of `php-install`.

> Did you test it also your charges with `nextcloud_install_websrv: true`?

Nope, sorry :innocent:  
I messed up with the file extension; this is fixed now, so CI should pass (testes locally with molecule).

**Comment by bemyak** [comment]
The last commit adds `php-bz2` module as the [Nextcloud docs](https://docs.nextcloud.com/server/latest/admin_manual/installation/source_installation.html) say that it's `recommended, required for extraction of apps`

File: tasks/websrv_install.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- name: "[INSTALL] - Apache Required package is installed."
+  ansible.builtin.package:
+    name: "libapache2-mod-php{{ php_ver }}"
+    state: present
+  when: nextcloud_websrv == "apache2"
+  notify:
+    - Start http
+
+- name: "[INSTALL] - NGINX Required package is installed."
```
Comment (staticdev) [inline tasks/websrv_install.yml:17]:
```suggestion
- name: "[INSTALL] - NGINX required package is installed."
```

**Review by staticdev — CHANGES_REQUESTED** [review]
Thanks again for your PR @bemyak I had some time to review it and has some minor findings.

File: tasks/php_install.yml [hunk 2]
```diff
@@ -1,43 +1,27 @@
...
-- name: "[INSTALL] -  Required and recommended packages are installed."
+- name: "[INSTALL] - Required and recommended packages are installed."
   ansible.builtin.package:
     name: "{{ item }}"
     state: present
   with_items:
-    - "{{ nextcloud_websrv }}"
     - imagemagick
     - smbclient
+    - "php-fpm"
```
[the remark this answers is not in the document]
Comment (bemyak) [reply tasks/php_install.yml:9]: On which distro are you checking? On Ubuntu 22.04 it works without the version:
```
root@nextcloud:~# apt install php-fpm
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
php-fpm is already the newest version (2:8.1+92ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 17 not upgraded.
```

But I will add it, no problem.

**Review by wiktor2200 — CHANGES_REQUESTED** [review]

File: tasks/php_install.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (staticdev) [reply tasks/php_install.yml]: Debian 11

File: tasks/main.yml [hunk 3]
```diff
@@ -5,7 +5,11 @@
   ansible.builtin.include_tasks: ./setup_env.yml
   tags: always
 
-- name: "Install required packages"
+- name: Install web server packages
```
[the remark this answers is not in the document]
Comment (staticdev) [reply tasks/main.yml:8]: @wiktor2200 oh, I missed that one!

**Comment by bemyak** [comment]
Thanks for the review! The issues are addressed now.

**Review by staticdev — APPROVED** [review]

**Comment by staticdev** [comment]
@wiktor2200 LGTM?

**Comment by wiktor2200** [comment]
Looks good now :)

śr., 28 gru 2022, 14:46 użytkownik staticdev ***@***.***>
napisał:

**Comment by wiktor2200** [comment]
Looks good now :)

**Review by wiktor2200 — APPROVED** [review]

**Pull request merged**